.github/workflows/test-and-release.yml

```diff
@@ -0,0 +1,111 @@
+on:
+  push:
+    branches:
+      - main
+      - next
+    tags:
+      - '*'
+  pull_request:
+permissions:
+  contents: write
+jobs:
+  full:
+    name: Node.js Latest Full
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@v4
+      - name: Install pnpm
+        uses: pnpm/action-setup@v2
+        with:
+          version: 8
+      - name: Install Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20
+          cache: pnpm
+      - name: Install dependencies
+        run: pnpm install --ignore-scripts
+      - name: Run tests
+        run: pnpm test
+  short:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version:
+          - 18
+          - 16
+    name: Node.js ${{ matrix.node-version }} Quick
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@v4
+      - name: Install pnpm
+        uses: pnpm/action-setup@v2
+        with:
+          version: 8
+      - name: Install Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v4
+        with:
+          node-version: ${{ matrix.node-version }}
+          cache: pnpm
+      - name: Install dependencies
+        run: pnpm install --ignore-scripts
+      - name: Run unit tests
+        run: pnpm unit
+  benchmark:
+    name: Benchmark
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@v4
+      - name: Install pnpm
+        uses: pnpm/action-setup@v2
+        with:
+          version: 8
+      - name: Install Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20
+          cache: pnpm
+      - name: Install dependencies
+        run: pnpm install --ignore-scripts
+      - name: Run benchmark
+        run: node ./test/benchmark.js
+  release:
+    name: Release On Tag
+    if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags/')
+    needs: [full, short, benchmark]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@v4
+      - name: Extract the changelog
+        id: changelog
+        run: |
+          TAG_NAME=${GITHUB_REF/refs\/tags\//}
+          READ_SECTION=false
+          CHANGELOG_CONTENT=""
+          while IFS= read -r line; do
+            if [[ "$line" =~ ^#+\ +(.*) ]]; then
+              if [[ "${BASH_REMATCH[1]}" == "$TAG_NAME" ]]; then
+                READ_SECTION=true
+              elif [[ "$READ_SECTION" == true ]]; then
+                break
+              fi
+            elif [[ "$READ_SECTION" == true ]]; then
+              CHANGELOG_CONTENT+="$line"$'\n'
+            fi
+          done < "CHANGELOG.md"
+          CHANGELOG_CONTENT=$(echo "$CHANGELOG_CONTENT" | awk '/./ {$1=$1;print}')
+          echo "changelog_content<<EOF" >> $GITHUB_OUTPUT
+          echo "$CHANGELOG_CONTENT" >> $GITHUB_OUTPUT
+          echo "EOF" >> $GITHUB_OUTPUT
```

Comment:
Can you show an example of description?

Reply:
Found in https://github.com/edloidas/automation

Reply:
Sure. It is in the demo repo I mentioned above. Check out [CHANGELOG.md](https://github.com/edloidas/automation/blob/main/CHANGELOG.md) and [releases](https://github.com/edloidas/automation/releases) with notes.